components/service/glean/src/main/java/mozilla/components/service/glean/utils/FreshBootDetector.kt

```diff
@@ -0,0 +1,118 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean.utils
+
+import android.content.Context
+import android.os.Build
+import android.provider.Settings
+import android.support.annotation.VisibleForTesting
+import java.io.File
+import java.io.IOException
+import java.util.UUID
+
+internal object FreshBootDetector {
```

Comment:
Can we have somebody from android components kindly review this? Seems very Android specific and I don't feel too comfortable in reviewing this :( @pocmo , any chance you could help ? :)

Reply:
This is somewhat scary. I have no clue how stable that is and if that works across devices. 🤷‍♂️ 

It would be interesting to collect telemetry about the boot id 😄